aiida/backends/djsite/db/migrations/0046_inline_calc_attributes.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Migrate attributes from legacy `InlineCalculation` nodes that were migrated to `CalcFunctionNode`s.
+The node type migration was performed in `0020_provenance_redesign.py` but the attributes were forgotten.
+
+* OLD `function_name` remain the same
+* OLD `namespace` becomes `function_namespace`
+* OLD `first_line_source_code` becomes `function_starting_line_number`
+* OLD `source_file` is moved to repository and dropped from DB
+* OLD `source_code` is used to calculate `function_number_of_lines` and dropped from DB
+* NEW `exit_status` is initialized at 0
+* NEW `process_state` is initialized at "finished"
+* NEW `process_label` is set as `Legacy InlineCalculation`
+
+Note: `exit_status` and `process_state` are initialized to 0 and "finished" respectively because the
+CalcFunctions affected are old InlineCalculation, which were only stored if they succeeded. You can check
+the relevant code here (line 60 has the execution call, it is stored after that):
+
+https://github.com/aiidateam/aiida-core/blob/v0.12.5/aiida/orm/implementation/django/calculation/inline.py#L60
+"""
+# pylint: disable=invalid-name
+from django.db import migrations, transaction
+
+from aiida.backends.general.migrations import utils
+from aiida.backends.djsite.db.migrations import upgrade_schema_version
+
+REVISION = '1.0.46'
+DOWN_REVISION = '1.0.45'
+
+
+def create_new_attributes(apps, _):
+    """Creates the new attributes"""
+    DbNode = apps.get_model('db', 'DbNode')
+
+    with transaction.atomic():
+        inline_type = 'process.calculation.calcfunction.CalcFunctionNode.'
+        inline_nodes = DbNode.objects.filter(node_type=inline_type, attributes__has_key='namespace')
+
+        for inline_node in inline_nodes:
```

Comment:
Looping and saving one by one will be slow. Probably, however, since this is only for legacy calculations, we can accept this, hoping that most DBs will have few of these entries... I don't know if the time to speed this up is worth the improvement, in this specific case.